### PR TITLE
Feature/map 209/ban user

### PR DESCRIPTION
**Pull request by Yojuks**

### CODE REVIEW

**Review by ghost — APPROVED** [review]

File: frontend/public/locales/ua/translation.json [hunk 1]
```diff
@@ -62,12 +62,15 @@
       "yourName": "Ваше ім'я:",
       "profilePageUpdatedSuccessfully": "Дані профілю успішно оновлені!",
       "subscribe": "Підписатись",
-      "unsubscribe": "Відписатись"
+      "unsubscribe": "Відписатись",
+      "ban": "забанити",
+      "unban": "відбанити"
     }
```
Comment (6mahdapihka9) [inline frontend/public/locales/ua/translation.json]: mb better "розбанити"?

File: frontend/src/components/PointPopup/IconsComponent/IconsComponent.tsx [hunk 2]
```diff
@@ -66,6 +66,9 @@ export const IconsComponent = ({
     state => state.isUserAuthorized.data
   );
   const { role } = useTypedSelector(state => state.isUserAuthorized.data);
+  const { role: OtherUserId } = useTypedSelector(
+    state => state.otherUserData.data
+  );
```
Comment (6mahdapihka9) [inline frontend/src/components/PointPopup/IconsComponent/IconsComponent.tsx]: why is the value of property 'role' assigned in the 'id' variable?

Comment (6mahdapihka9) [reply frontend/src/components/PointPopup/IconsComponent/IconsComponent.tsx]: should be renamed to 'otherUserRole' or smthg

File: frontend/src/redux/reducers/otherUserDataReducer.tsx [hunk 3]
```diff
@@ -67,6 +68,20 @@ export const otherUserDataReducer = (
...
+        success: false
+      };
+    case OtherUserDataActionTypes.UPDATE_USER_DATA_AND_BAN_SUCCESS:
+      return {
+        loading: false,
+        error: null,
+        data: action.payload,
+        success: true
+      };
     default:
```
Comment (6mahdapihka9) [inline frontend/src/redux/reducers/otherUserDataReducer.tsx]: i think 'onError' case should be added here

File: frontend/src/screens/PersonProfile/PersonProfilePage.tsx [hunk 4]
```diff
@@ -51,6 +58,27 @@ export default function PersonProfilePage() {
...
+
+  const handleUnban = () => {
+    if (isAuthorized) {
+      toggleUserBan(
+        otherUserId,
+        'user',
+        t('personalProfile.personalProfilePage.banningUpdated')
+      );
+    }
+  };
```
Comment (6mahdapihka9) [inline frontend/src/screens/PersonProfile/PersonProfilePage.tsx]: it should toggle between:
- 'banned' and currentUserRole
not:
- 'banned' and 'user'

mb I'm wrong, explain please

**Review by 6mahdapihka9 — CHANGES_REQUESTED** [review]

File: frontend/public/locales/ua/translation.json (hunk 1 above)
Comment (Yojuks) [reply frontend/public/locales/ua/translation.json]: mb - done)

File: frontend/src/components/PointPopup/IconsComponent/IconsComponent.tsx (hunk 2 above)
Comment (Yojuks) [reply frontend/src/components/PointPopup/IconsComponent/IconsComponent.tsx]: i agree with you - done

File: frontend/src/screens/PersonProfile/PersonProfilePage.tsx (hunk 4 above)
Comment (Yojuks) [reply frontend/src/screens/PersonProfile/PersonProfilePage.tsx]: done

Comment (Yojuks) [reply frontend/src/screens/PersonProfile/PersonProfilePage.tsx]: done

File: frontend/src/redux/reducers/otherUserDataReducer.tsx (hunk 3 above)
Comment (Yojuks) [reply frontend/src/redux/reducers/otherUserDataReducer.tsx]: done

**Review by mazursasha1990 — CHANGES_REQUESTED** [review]
There is an error when opening the popup location.
ClickLocationError
n

File: frontend/src/redux/reducers/otherUserDataReducer.tsx [hunk 5]
```diff
@@ -67,6 +68,27 @@ export const otherUserDataReducer = (
         error: action.payload,
         success: false
       };
+    case OtherUserDataActionTypes.UPDATE_USER_DATA_AND_BAN_LOADING:
+      return {
+        loading: false,
```
Comment (mazursasha1990) [inline frontend/src/redux/reducers/otherUserDataReducer.tsx]: This loading must be "true".

**Review by mazursasha1990 — CHANGES_REQUESTED** [review]

File: frontend/src/redux/reducers/otherUserDataReducer.tsx (hunk 5 above)
Comment (Yojuks) [reply frontend/src/redux/reducers/otherUserDataReducer.tsx]: done

**Review by mazursasha1990 — APPROVED** [review]

**Review by 6mahdapihka9 — APPROVED** [review]

**Pull request merged**